warrior/test_WarriorCore/test_common_execution_utils.py

```diff
@@ -0,0 +1,182 @@
+    del common_execution_utils.get_object_from_datarepository
+    del common_execution_utils.getAbsPath
+
+
+# def test_get_step_list_for_loop_tag():
+#     #data_Utils.update_datarepository = MagicMock(return_value={})
+
+#     file_path = os.path.join(os.path.split(__file__)[0], "test_loop_datatype.xml")
+#     json_file_path = os.path.join(os.path.split(__file__)[0], "1j_loop_datatype.json")
+#     common_execution_utils.update_datarepository = MagicMock(return_value={})
+#     common_execution_utils.get_object_from_datarepository = MagicMock(return_value=file_path)
+#     common_execution_utils.getAbsPath = MagicMock(return_value=json_file_path)
+#     common_execution_utils.getAbsPath = MagicMock(return_value=json_file_path)
+#     common_execution_utils.get_step_list(filepath=file_path,  step_tag="Steps", sub_step_tag="step", randomize=True )
+
+def test_get_runmode_from_xmlfile():
+    '''test_get_runmode_from_xmlfile'''
+    tree = ET.parse(os.path.join(os.path.split(__file__)[0], "sample.xml"))
+    # get root element
+    root = tree.getroot()
+    # getting steps
+    steps = root.find("Steps")
+    sstep = steps[0]
+    value = 4
+    go_next = 5
+    mode = 'runmode'
+    tag = 'value'
+
+    common_execution_utils.get_runmode_from_xmlfile(element=sstep)
+
+def test_get_runmode_from_xmlfile_no_runmode_timer():
+    '''test_get_runmode_from_xmlfile_no_runmode_timer'''
+    tree = ET.parse(os.path.join(os.path.split(__file__)[0], "sample.xml"))
+    # get root element
+    root = tree.getroot()
+    # getting steps
+    steps = root.find("Steps")
+    sstep = steps[1]
+    common_execution_utils.get_runmode_from_xmlfile(element=sstep)
+
+def test_get_runmode_from_xmlfile_invalid_runmode_timer():
+    '''test_get_runmode_from_xmlfile_invalid_runmode_timer'''
+    tree = ET.parse(os.path.join(os.path.split(__file__)[0], "sample.xml"))
+    # get root element
+    root = tree.getroot()
+    # getting steps
+    steps = root.find("Steps")
+    sstep = steps[2]
+    common_execution_utils.get_runmode_from_xmlfile(element=sstep)
+
+def test_get_runmode_from_xmlfile_invalid_runmode_type():
+    '''test_get_runmode_from_xmlfile_invalid_runmode_type'''
+    tree = ET.parse(os.path.join(os.path.split(__file__)[0], "sample.xml"))
+    # get root element
+    root = tree.getroot()
+    # getting steps
+    steps = root.find("Steps")
+    sstep = steps[3]
+    status = common_execution_utils.get_runmode_from_xmlfile(element=sstep)
+    assert status[0] == None
+
+def test_get_runmode_from_xmlfile_runmode_value_as_zero():
+    '''test_get_runmode_from_xmlfile_runmode_value_as_zero'''
+    tree = ET.parse(os.path.join(os.path.split(__file__)[0], "sample.xml"))
+    # get root element
+    root = tree.getroot()
+    # getting steps
+    steps = root.find("Steps")
+    sstep = steps[4]
+    status = common_execution_utils.get_runmode_from_xmlfile(element=sstep)
+    assert status[0] == "RMT"
```

Comment:
Add two more asserts here .
refer the source code for the same .
https://github.com/warriorframework/warriorframework_py3/blob/master/warrior/WarriorCore/common_execution_utils.py#L260